File: src/decorator/python/EmailNotificationSenderDecorator.py
from src.decorator.python.NotificationSenderDecorator import NotificationSenderDecorator


class EmailNotificationSenderDecorator(NotificationSenderDecorator):
    def sendNotification(self):
        print("Gotta send email notification")
        self.sender.sendNotification()
        print("Email notification sent!")

File: src/decorator/java/com/my/EmailNotificationSenderDecorator.java
package src.decorator.java.com.my;

public class EmailNotificationSenderDecorator extends NotificationSenderDecorator {
    public EmailNotificationSenderDecorator(NotificationSender sender) {
        super(sender);
    }

    @Override
    public void sendNotification() {
        System.out.println("Gotta send email notification");
        super.sendNotification();
        System.out.println("email notification sent!");
    }
}
